File: TaggyAppBackend/TaggyAppBackend.Api/Models/Dtos/File/CreateFileDto.cs
using System.ComponentModel.DataAnnotations;
using TaggyAppBackend.Api.Models.Dtos.Tag;

namespace TaggyAppBackend.Api.Models.Dtos.File;

public class CreateFileDto
{
    [Required][MaxLength(255)] public string Name { get; set; } = null!;
    public string? Description { get; set; }
    public IEnumerable<CreateTagDto> Tags { get; set; } = [];
}

File: TaggyAppFrontend/src/app/models/dtos/file/createFileDto.ts
import { maxLength, minLength, required } from '@rxweb/reactive-form-validators';
import { CreateTagDto } from '../tag/createTagDto';

export class CreateFileDto {
  @required()
  @minLength({ value: 3 })
  @maxLength({ value: 255 })
  untrustedName!: string;

  @maxLength({ value: 255 })
  description?: string;

  tags: CreateTagDto[] = [];
}
